Add RWeka Tuning Parameters

dashaub opened the pull request:
Adding the tuning parameters as discussed in issue #468 . Some notes:
- See issue #468 for the model performance results of before and after.
- Tuning grids for many models did not implement a random search previously. Random search is now implemented, and the grid search is also improved to test more than a single model.
- Tuning grids for PART and J48 only tested `C = 0.25` previously. We now sample in `C` in [0, 0.5].
- Sampling for many tuning parameters is set to a maximum `upperBound <- min(50, nrow(data) / 2) ` to prevent models failing to train; even for very large `nrow(data)`, performance when the tuning parameters were set to >50 was uniformly poor. Furthermore, testing showed that although this was the upper possible values for which the model would train, optimal models usually had much lower values, so sampling is done on the log scale instead of uniform `1:upperBound` to increase the probability of each model fit being in a good region for that tuning parameter.
- I ran `parseModels.R` for testing, but the updated `models.RData` object isn't included in this PR, so you'll have to run this on your end.
- The grid search construction for JRip is a little funny since we need to sample integer values on three tuning parameters. If you can think of a cleaner way to implement sampling on an integer grid here, go for it. Ultimately I decided to sample effectively on two of them and vary `NumFolds = 2:4` since that is usually a good range for `NumFolds`.
- "Collisions" for some of the random tuning grids generated on whole numbers--not a problem when we're sampling a continuous instead of a discrete numbers--are resolved by generating additional models, selecting `unique()` models, and limiting to the desired `len` with subsetting.

topepo commented:
One request... when using `J48`, can you add some logic to make `C = .25` when `tuneLength = 1`? If would be a good idea to enable people to use the default that Quinlan suggests easily.

dashaub commented:
Good idea. I also updated the default values for `PART` and `JRip` to their WEKA defaults for when `tuneLength = 1` and fixed the grid sorting to sort on the added tuning parameters.

dashaub commented:
Added some fixes and cleanup to the grid search tunelength.
